control-plane/pkg/reconciler/trigger/namespaced_controller.go

```diff
@@ -0,0 +1,132 @@
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/base"
+	"knative.dev/eventing/pkg/apis/feature"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	brokerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/broker"
+	triggerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/trigger"
+	triggerreconciler "knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1/trigger"
+)
+
+// TODO: COPY & PASTE for now
+
+const (
+	NamespacedControllerAgentName = "kafka-namespaced-trigger-controller"
+
+	NamespacedFinalizerName = "kafka.namespaced.triggers.eventing.knative.dev"
+)
+
+func NewNamespacedController(ctx context.Context, watcher configmap.Watcher, configs *config.Env) *controller.Impl {
+
+	logger := logging.FromContext(ctx).Desugar()
+
+	configmapInformer := configmapinformer.Get(ctx)
+	brokerInformer := brokerinformer.Get(ctx)
+	triggerInformer := triggerinformer.Get(ctx)
+	triggerLister := triggerInformer.Lister()
+
+	reconciler := &NamespacedReconciler{
+		Reconciler: &base.Reconciler{
+			KubeClient:                  kubeclient.Get(ctx),
+			PodLister:                   podinformer.Get(ctx).Lister(),
+			SecretLister:                secretinformer.Get(ctx).Lister(),
+			DataPlaneConfigMapNamespace: configs.DataPlaneConfigMapNamespace,
+			DataPlaneConfigMapName:      configs.DataPlaneConfigMapName,
+			DataPlaneConfigFormat:       configs.DataPlaneConfigFormat,
+			DataPlaneNamespace:          configs.SystemNamespace,
+			DispatcherLabel:             base.BrokerDispatcherLabel,
+			ReceiverLabel:               base.BrokerReceiverLabel,
+		},
+		BrokerLister:   brokerInformer.Lister(),
+		EventingClient: eventingclient.Get(ctx),
+		Env:            configs,
+		Flags:          feature.Flags{},
+	}
+
+	impl := triggerreconciler.NewImpl(ctx, reconciler, func(impl *controller.Impl) controller.Options {
+		return controller.Options{
+			FinalizerName:     NamespacedFinalizerName,
+			AgentName:         NamespacedControllerAgentName,
+			SkipStatusUpdates: false,
+			PromoteFilterFunc: filterTriggers(reconciler.BrokerLister, kafka.NamespacedBrokerClass, NamespacedFinalizerName),
+		}
+	})
+
+	featureStore := feature.NewStore(
+		logging.FromContext(ctx).Named("feature-config-eventing-store"),
+		func(name string, value interface{}) {
+			flags, ok := value.(feature.Flags)
+			if !ok {
+				logger.Warn("Features ConfigMap " + name + " updated but we didn't get expected flags. Skipping updating cached features")
+			}
+			logger.Debug("Features ConfigMap " + name + " updated. Updating cached features.")
+			reconciler.FlagsLock.Lock()
+			defer reconciler.FlagsLock.Unlock()
+			reconciler.Flags = flags
+			impl.GlobalResync(triggerInformer.Informer())
+		},
+	)
+	featureStore.WatchConfigs(watcher)
+
+	reconciler.Resolver = resolver.NewURIResolverFromTracker(ctx, impl.Tracker)
+
+	triggerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
```

Comment:
we need additional handlers for all additional resources we're creating

Reply:
wrong place for this comment, it is meant to go for the namespace_controller.go in broker package